testnet/launcher/docker.go

```diff
@@ -0,0 +1,243 @@
+			l2cd.WithDockerImage("testnetobscuronet.azurecr.io/obscuronet/hardhatdeployer:latest"),
+		),
+	)
+	if err != nil {
+		return fmt.Errorf("unable to configure the l2 contract deployer - %w", err)
+	}
+
+	err = l2ContractDeployer.Start()
+	if err != nil {
+		return fmt.Errorf("unable to start the l2 contract deployer - %w", err)
+	}
+	fmt.Println("L2 Contracts were successfully deployed...")
+
+	return nil
+}
+
+func startEth2Network() error {
+	eth2Network, err := eth2network.NewDockerEth2Network(
+		eth2network.NewEth2NetworkConfig(
+			eth2network.WithGethHTTPStartPort(8025),
+			eth2network.WithGethWSStartPort(9000),
+			eth2network.WithGethPrefundedAddrs([]string{
+				"0x13E23Ca74DE0206C56ebaE8D51b5622EFF1E9944", // contract deployment pk - f52e5418e349dccdda29b6ac8b0abe6576bb7713886aa85abea6181ba731f9bb
+				"0x0654D8B60033144D567f25bF41baC1FB0D60F23B", // sequencer pk - 8ead642ca80dadb0f346a66cd6aa13e08a8ac7b5c6f7578d4bac96f5db01ac99
+				"0x2f7fCaA34b38871560DaAD6Db4596860744e1e8A", // validator pk - ebca545772d6438bbbe1a16afbed455733eccf96157b52384f1722ea65ccfa89
+			}),
+		),
+	)
+	if err != nil {
+		return fmt.Errorf("unable to configure eth2network - %w", err)
+	}
+
+	err = eth2Network.Start()
+	if err != nil {
+		return fmt.Errorf("unable to start eth2network - %w", err)
+	}
+	fmt.Println("Eth2 network started...")
+
+	err = eth2Network.IsReady()
+	if err != nil {
+		return fmt.Errorf("eth2network not ready in time - %w", err)
+	}
+	fmt.Println("Eth2 network is ready...")
+
+	return nil
+}
+
+func deployL1Contracts() (string, string, error) {
+	l1ContractDeployer, err := l1cd.NewDockerContractDeployer(
+		l1cd.NewContractDeployerConfig(
+			l1cd.WithL1Host("eth2network"),
+			l1cd.WithL1Port(8025),
+			l1cd.WithPrivateKey("f52e5418e349dccdda29b6ac8b0abe6576bb7713886aa85abea6181ba731f9bb"),
+			l1cd.WithDockerImage("testnetobscuronet.azurecr.io/obscuronet/hardhatdeployer:latest"),
+		),
+	)
+	if err != nil {
+		return "", "", fmt.Errorf("unable to configure l1 contract deployer - %w", err)
+	}
+
+	err = l1ContractDeployer.Start()
+	if err != nil {
+		return "", "", fmt.Errorf("unable to start l1 contract deployer - %w", err)
+	}
+
+	managementContractAddr, messageBusContractAddr, err := l1ContractDeployer.RetrieveL1ContractAddresses()
+	if err != nil {
+		return "", "", fmt.Errorf("unable to fetch l1 contract addresses - %w", err)
+	}
+	fmt.Println("L1 Contracts were successfully deployed...")
+	return managementContractAddr, messageBusContractAddr, nil
+}
+
+// waitForHealthyNode retries continuously for the node to respond to a healthcheck http request
+func waitForHealthyNode(port int) error { // todo: hook the cfg
+	requestURL := fmt.Sprintf("http://localhost:%d", port)
+	reqBody := `{"method": "obscuro_health", "id": 1}`
+
+	fmt.Println("Waiting for obscuro node to be healthy...")
+	for startTime := time.Now(); time.Now().Before(startTime.Add(2 * time.Minute)); time.Sleep(time.Second) {
```

Comment:
I haven't checked but I imagine this will come out nicer with the `retry.Do()` thing and moving some of the low level processing into a separate function. You've got a nestif ignore and I imagine you're getting close to needing a complexity ignore too.

Reply:
I've actually used it previously, it's a good package to have, just forgot it :) Added it now .
I think it actually makes the code cleaner.